### PR TITLE
ECF-97 DFE admin login

### PR DESCRIPTION
**Pull request by amazimbe**
### Changes proposed in this pull request
A DfE administrator can log in with their email address
They can see an administrator dashboard on successful login

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
 # frozen_string_literal: true
 
 module ApplicationHelper
+  def profile_dashboard_url(user)
```
Comment (amazimbe) [inline app/helpers/application_helper.rb:4]: Just generating the right dashboard url based on the user's profile.

File: app/controllers/admin_dashboard_controller.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AdminDashboardController < ApplicationController
+  before_action :authenticate_user!
```
Comment (kicferk1) [inline app/controllers/admin_dashboard_controller.rb:4]: Can non-admin users access this dashboard? Do we need an extra check saying 'if user not admin, kick them out'?

Comment (kicferk1) [reply app/controllers/admin_dashboard_controller.rb:4]: Probably worth adding a test describing this behaviour too.

Comment (amazimbe) [reply app/controllers/admin_dashboard_controller.rb:4]: @kicferk1 that did cross my mind but I felt it was out of scope for this login pr. I'd typically use pundit for that and add policies and policy tests. I won't add them to this pr but I can do a separate one after this on when we beef up this controller.

Comment (kicferk1) [reply app/controllers/admin_dashboard_controller.rb:4]: Alright, happy for it to be a separate PR

**Review by kicferk1 — APPROVED** [review]
Should have approved yesterday, sorry if it delayed you :/

**Comment by ethanmills** [comment]
Could we seed an admin account for deployed-dev using a [mailinator ](https://www.mailinator.com/) email address? (and add that to the notify permitted list)

**Comment by amazimbe** [comment]
> Could we seed an admin account for deployed-dev using a [mailinator ](https://www.mailinator.com/) email address? (and add that to the notify permitted list)

@ethanmills I'm probably missing something. What do we need mailinator for?

**Comment by ethanmills** [comment]
> > Could we seed an admin account for deployed-dev using a [mailinator ](https://www.mailinator.com/) email address? (and add that to the notify permitted list)
> 
> @ethanmills I'm probably missing something. What do we need mailinator for?

Doesn't have to be that, I just thought it might make a convenient joint mailbox.

I would like to be able to log in as different user types on dev, but we can't all add lots of different emails addresses, because notify limits the number of permitted email addresses. Therefore I think it might be good to have some shared test accounts. That means we need some shared mailboxes.

**Comment by kicferk1** [comment]
Do we already have shared email accounts? Like `ecf-dev@digital.education.gov.uk` that we use for accessing paas when deploying?

**Comment by ethanmills** [comment]
That's a good point, I'd forgotten about those

**Comment by ethanmills** [comment]
The advantage of mailinator is that we'd be able to easily share it with the wider team, though.

File: db/seeds.rb [hunk 3]
```diff
@@ -13,3 +13,11 @@
 unless LeadProvider.first
   LeadProvider.create!(name: "Test Lead Provider")
 end
+
+unless AdminProfile.first
```
Comment (ethanmills) [inline db/seeds.rb:17]: This is great, exactly what I was thinking of. We potentially want this as 
```
unless AdminProfile.first  || Rails.env.production?
```
however

**Review by ethanmills — APPROVED** [review]

**Pull request merged**